prime-router/src/main/kotlin/azure/WorkflowEngine.kt

```diff
@@ -770,7 +776,11 @@ class WorkflowEngine(
 
         val downloadContent = (reportFile.bodyUrl != null && fetchBlobBody)
         val content = if (downloadContent && BlobAccess.exists(reportFile.bodyUrl)) {
-            BlobAccess.downloadBlob(reportFile.bodyUrl)
+            val startDownloadTime = Calendar.getInstance().timeInMillis
+            val contentToReturn = BlobAccess.downloadBlob(reportFile.bodyUrl)
```

Comment:
Downloading data in a factory method is a weird thing to do imo, have we tried to see what it would take to refactor this?